aw_client/client.py

```diff
@@ -129,65 +127,110 @@ def disconnect(self):
         # FIXME: doesn't disconnect immediately
         self.dispatch_thread.running = False
 
+QueuedRequest = namedtuple("QueuedRequest", ["endpoint", "data"])
+
+
+def RestartOnException(f):
+    @functools.wraps(f)
+    def g(*args, **kwargs):
+        while True:
+            try:
+                f(*args, **kwargs)
+            except Exception as e:
+                logger.error("{} crashed due to exception, restarting.".format(f))
+                logger.error(e)
+                time.sleep(1)  # To prevent extremely fast restarts in case of bad state.
+    return g
+
 
 class PostDispatchThread(threading.Thread):
     def __init__(self, client):
-        threading.Thread.__init__(self)
-        self.daemon = True
+        threading.Thread.__init__(self, daemon=True)
         self.running = True
         self.connected = False
+
         self.client = client
-        self.queue = Queue()
+        self._queue = Queue()
+
+        # Setup failed queues file
+        data_dir = get_data_dir("aw-client")
+        failed_queues_dir = os.path.join(data_dir, "failed_requests")
+        if not os.path.exists(failed_queues_dir):
+            os.makedirs(failed_queues_dir)
+        self.queue_file = os.path.join(failed_queues_dir, self.client.client_name)
+
         self._load_queue()
```

Comment:
This line should be completely removed to fix clearing the file queue when not intended. My mistake